index.html

```diff
@@ -4292,6 +4306,454 @@ <h3>Proof Formats</h3>
+`application/vp+ld+json`, the <var>extractDocument</var> function takes
+<var>inputBytes</var> as an input parameter, parses the value as JSON,
+and returns the result as an object. If JSON parsing fails, a
+<a href="#PARSING_ERROR">PARSING_ERROR</a> is appended to
+<var>securingMechanismVerified</var>.<var>errors</var>.
+              </li>
+              <li>
+If the <var>inputMediaType</var> is associated with the JOSE or COSE
+specifications, the [[?VC-JOSE-COSE]] specification defines a process for
+extracting the <a>conforming document</a>.
+              </li>
+              <li>
+Other extracting functions are possible, but need not be listed in
+the <a data-cite="?VC-SPECS#media-type-extensions">Media Type Extensions</a>
+section of the Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+Implementers might find these other specifications through other standards
+bodies or communities that utilize this specification.
+              </li>
+            </ol>
+          </li>
+          <li>
+Determine the specific cryptographic verification algorithm
+(<var>verifyProof</var>) required to verify the securing mechanism:
+            <ol class="algorithm">
+              <li>
+If the <var>inputMediaType</var> is `application/vc+ld+json` or
+`application/vp+ld+json`, the Verifiable Credential Data Integrity
+Specification [[?VC-DATA-INTEGRITY]] defines the process for identifying
+a specific cryptography suite type used for cryptographic verification by
+utilizing the `proof`.`type` property. The list of all registered
+proof types can be found in the <a data-cite="?VC-SPECS#proof">Proof Types</a>
+section of the Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+              </li>
+              <li>
+If the <var>inputMediaType</var> is associated with the JOSE or COSE
+specifications, the [[?VC-JOSE-COSE]] specification defines a process for
+identification of a specific cryptography suite and cryptographic verification
+process.
+              </li>
+              <li>
+Other proof verification functions are possible, but need not be listed in
+the <a data-cite="?VC-SPECS#proof">Proof Types</a> section of the
+Verifiable Credentials Specifications Directory [[?VC-SPECS]].
+Implementers might find these other specifications through other standards
+bodies or communities that utilize this specification.
+              </li>
+            </ol>
+          </li>
+          <li>
+Set <var>securingMechanismVerified</var>.<var>status</var> to the
+result of passing <var>inputBytes</var>, along with any relevant function-specific
+options, to the <var>verifyProof</var> function. For example, for
+Verifiable Credential Data Integrity [[?VC-DATA-INTEGRITY]] proof
+verification functions, the <var>expectedProofPurpose</var> option is set to
+`assertionMethod`, and if the <a>conforming document</a> is expected to be a
+<a>verifiable presentation</a>, the <var>domain</var> and <var>challenge</var>
+are also provided as options.
```

Comment:
This algorithm should specify the relevant function-specific options (probably in the parts that dispatch based on the media type and proof type) rather than making implementations guess.

Reply:
I have made an attempt at fixing this here: bc06e7d108d805a77cb1a9c968504f18a1a073e2

It makes the algorithm more verbose than I'd like, and that verbosity could be reduced if we define a common interface across all securing mechanisms, but the WG will have to agree to doing that. In the meantime, I think the text that's there works (even though it's more verbose than we'd prefer).